### PR TITLE
Bug/gb 56 oncomplete called before finished

### PR DESCRIPTION
**Pull request by rfreitas**
Fix for issue: https://github.com/onfido/onfido-sdk-ui/issues/56

Must be reviewed together with: https://github.com/onfido/onfido-sdk-core/pull/18

### CODE REVIEW

**Comment by stephencookdev** [comment]
What was it in the old code that was actually causing the bug? I can't see it

**Comment by rfreitas** [comment]
Forgot to include the complementary PR: https://github.com/onfido/onfido-sdk-core/pull/18

File: package.json [hunk 1]
```diff
@@ -7,14 +7,17 @@
     "dev": "NODE_ENV=development webpack-dev-server --inline --hot --progress --https",
     "start": "superstatic build -p ${PORT:-8080} --host 0.0.0.0 --gzip -c '{\"rewrites\": [{\"source\":\"**\",\"destination\":\"index.html\"}],\"headers\":[{\"source\":\"**\",\"headers\":[{\"key\":\"Cache-Control\",\"value\":\"max-age=86400\"}]}]}'",
     "prestart": "npm run build",
-    "build": "NODE_ENV=production webpack -p --progress",
-    "prebuild": "npm run clean && mkdir -p dist && mkdir -p lib",
+    "build": "NODE_ENV=production webpack -p --progress ",
+    "build:core": "(cd ../onfido-sdk-core && npm run build)",
```
Comment (stephencookdev) [inline package.json]: This feels really hacky... There must be a better way of doing this?

File: package.json [hunk 2]
```diff
@@ -7,14 +7,17 @@
...
+    "prebuild": "npm run clean && mkdir -p dist && mkdir -p lib && (npm run build:core || true)",
     "clean": "rimraf lib dist",
     "check:src": "npm run lint && npm run test",
     "prepublish": "npm run clean && npm run check:src && npm run build",
     "test": "npm run -s lint && npm run -s test:karma",
     "test:karma": "karma start test/karma.conf.js --single-run",
-    "lint": "eslint {src,test}"
+    "lint": "eslint {src,test}",
+    "link:core": "npm link ../onfido-sdk-core",
```
Comment (stephencookdev) [inline package.json]: Is this really something we should be putting in npm scripts?

`npm link` is pretty standard, and this script makes loads of assumptions about where `onfido-sdk-core` is etc.

File: package.json [hunk 3]
```diff
@@ -7,14 +7,17 @@
...
+    "unlink:core": "rm -r ./node_modules/onfido-sdk-core && npm install"
```
Comment (stephencookdev) [inline package.json]: Probably just `npm install onfido-sdk-core`, if we are to keep this. Would be faster

**Comment by stephencookdev** [comment]
Other than my reasonably minor comments, LGTM 👍

File: package.json (hunk 1 above)
Comment (rfreitas) [inline package.json]: definitely a total hack, but I needed something at least :/

File: package.json (hunk 2 above)
Comment (rfreitas) [inline package.json]: what I wanted was simply speed when linking and unlinking the core, but I get the argument, I might drop the npm link one.

File: package.json (hunk 3 above)
Comment (rfreitas) [inline package.json]: good point

**Pull request merged**